R/compare_designs.R

```diff
@@ -0,0 +1,216 @@
+#' Compare two designs
+#'
+#' @param design1 A design object, typically created using the + operator
+#' @param design2 A design object, typically created using the + operator
+#' @param format Format (in console or HTML) options from \code{diffobj::diffChr}
+#' @param mode Mode options from \code{diffobj::diffChr}
+#' @param pager Pager option from \code{diffobj::diffChr}
+#' @param context Context option from \code{diffobj::diffChr}
+#' @param x A design comparison object created by \code{compare_designs}
+#' @param ... Options for printing design comparison object
+#'
+#' @examples
+#' 
+#' design1 <- declare_population(N = 100, u = rnorm(N)) +
+#'   declare_potential_outcomes(Y ~ Z + u) +
+#'   declare_estimand(ATE = mean(Y_Z_1 - Y_Z_0)) +
+#'   declare_sampling(n = 75) +
+#'   declare_assignment(m = 50) +
+#'   declare_reveal(Y, Z) +
+#'   declare_estimator(Y ~ Z, estimand = "ATE")
+#' 
+#' design2 <- declare_population(N = 200, u = rnorm(N)) +
+#'   declare_potential_outcomes(Y ~ 0.5*Z + u) +
+#'   declare_estimand(ATE = mean(Y_Z_1 - Y_Z_0)) +
+#'   declare_sampling(n = 100) +
+#'   declare_assignment(m = 25) +
+#'   declare_reveal(Y, Z) +
+#'   declare_estimator(Y ~ Z, model = lm_robust, estimand = "ATE")
+#'  
+#'  compare_designs(design1, design2)
+#'  compare_design_code(design1, design2)
+#'  compare_design_summaries(design1, design2)
+#'  compare_design_data(design1, design2)
+#'  compare_design_estimates(design1, design2)
+#'  compare_design_estimands(design1, design2)
+#' 
+#' @name compare_functions
+
+#' @rdname compare_functions
+#' @export
+compare_designs <- function(design1, design2, format = "ansi8", mode = "sidebyside", pager = "off", context = -1L) {
+  
+  code_comparison <- compare_design_code(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  data_comparison <- compare_design_data(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  estimands_comparison <- compare_design_estimands(design1, design2, format = format, mode = mode, pager = pager, context = context)
+  
+  estimates_comparison <- compare_design_estimates(design1, design2, format = format, mode = mode, pager = pager, context = context)
```

Comment:
eek. Use a list to cut down on verbosity.

    fns <- list(code_comparison=compare_design_code, data_comparison=compare_design_data, ...)

    vals <- lappy(fns, function(f) f(design1, design2, format=format, mode=mode, pager=pager, context=context)

    class(vals) <- "design_comparison"
    vals